packages/tools/src/tools/VolumeCroppingControlTool.ts

```diff
@@ -0,0 +1,1771 @@
+          intersections.push({
+            with: otherLine[3], // 'min' or 'max'
+            point: intersection,
+          });
+        }
+      }
+
+      // get color for the reference line
+      const otherViewport = line[0];
+      const viewportColor = this._getReferenceLineColor(otherViewport.id);
+      const viewportControllable = this._getReferenceLineControllable(
+        otherViewport.id
+      );
+      const selectedViewportId = data.activeViewportIds.find(
+        (id) => id === otherViewport.id
+      );
+
+      let color =
+        viewportColor !== undefined ? viewportColor : 'rgb(200, 200, 200)';
+
+      let lineWidth = 2.5;
+
+      const lineActive =
+        data.handles.activeOperation !== null &&
+        data.handles.activeOperation === OPERATION.DRAG &&
+        selectedViewportId;
+
+      if (lineActive) {
+        lineWidth = 4.5;
+      }
+
+      const lineUID = `${lineIndex}`;
+      if (viewportControllable) {
+        if (intersections.length === 2) {
+          drawLineSvg(
+            svgDrawingHelper,
+            annotationUID,
+            lineUID,
+            intersections[0].point,
+            intersections[1].point,
+            //   line[1],
+            //  line[2],
+            {
+              color,
+              lineWidth,
+            }
+          );
+        }
+      }
+
+      if (viewportControllable) {
+        color =
+          viewportColor !== undefined ? viewportColor : 'rgb(200, 200, 200)';
+        if (lineActive) {
+          const handleUID = `${lineIndex}`;
+        }
+      }
+    });
+
+    renderStatus = true;
+
+    if (this.configuration.viewportIndicators) {
+      const { viewportIndicatorsConfig } = this.configuration;
+      const xOffset = viewportIndicatorsConfig?.xOffset || 0.95;
+      const yOffset = viewportIndicatorsConfig?.yOffset || 0.05;
+      const referenceColorCoordinates = [
+        clientWidth * xOffset,
+        clientHeight * yOffset,
+      ];
+
+      const circleRadius =
+        viewportIndicatorsConfig?.circleRadius || canvasDiagonalLength * 0.01;
+
+      const circleUID = '0';
+      drawCircleSvg(
+        svgDrawingHelper,
+        annotationUID,
+        circleUID,
+        referenceColorCoordinates as Types.Point2,
+        circleRadius,
+        { color, fill: color }
+      );
+    }
+
+    return renderStatus;
+  };
+
+  _getAnnotations = (enabledElement: Types.IEnabledElement) => {
+    const { viewport } = enabledElement;
+    const annotations =
+      getAnnotations(this.getToolName(), viewport.element) || [];
+    const viewportIds = this._getViewportsInfo().map(
+      ({ viewportId }) => viewportId
+    );
+
+    // filter the annotations to only keep that are for this toolGroup
+    const toolGroupAnnotations = annotations.filter((annotation) => {
+      const { data } = annotation;
+      return viewportIds.includes(data.viewportId);
+    });
+
+    return toolGroupAnnotations;
+  };
+
+  _onSphereMoved = (evt) => {
```

Comment:
The logic for updating min/max/corner spheres is a bit hard to follow and could be error-prone if the indices change. Consider using named constants or enums for clarity.